### ISSUE TITLE
Mechanism to provide more detailed progress information and error reporting for long-running operations

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#4236]
OpenRefine offers a few long-running operations:
* Add a column by fetching URLs
* Reconcile
* Add a column from reconciled values (aka. data extension)
* Upload edits to Wikibase

Since these operations can take a long time to be executed, and can encounter errors (web service not reachable, rate-limiting events…) it would be good to have more feedback than a progress percentage. We currently lack a uniform mechanism for all these operations to expose this progress.

### Proposed solution

It is not clear to me what this should look like.
The progress statistics could be expanded to include:
* Estimated time of completion?
* Number of errors encountered so far? (So that users can cancel the operation early if they see a large number of errors there)
* Processing speed (number of rows/records per second?)

For error reporting, the "Add column by fetching URLs" give the option to store errors as cell values. This could be extended to other long-running operations:
* The reconcile operation could be given the option to store errors in a separate column, or add the ability to store errors in the same column without overriding the existing values (https://github.com/OpenRefine/OpenRefine/issues/3369#issuecomment-947677065).
* Adding columns from reconciled values could also be given the option to store errors in a dedicated column
* Uploading edits to Wikibase: that is more difficult, since it is not so easy to trace back an edit to a row / record (some edits may not come from any row, some others come from all rows… Potentially it could create another project, with one column for the edited entity, another for the time of the attempted edit, and one for the error encountered? But that would quickly pollute workspaces (creating a lot of projects to store these errors, perhaps.

### Additional context

Generalization of #3369.

**Comment by wetneb** [comment]
Error handling is particularly a problem for our Wikibase upload operation, I feel. Edits on a Wikibase instance can fail for a lot of different reasons and we are not doing a good job of exposing those failures to the user.

This will be all the more important for Wikimedia Commons, since file uploads can also fail for even more reasons.
@lozanaross if you have any thoughts :)

**Comment by lozanaross** [comment]
> Error handling is particularly a problem for our Wikibase upload operation, I feel. Edits on a Wikibase instance can fail for a lot of different reasons and we are not doing a good job of exposing those failures to the user.
> 
> This will be all the more important for Wikimedia Commons, since file uploads can also fail for even more reasons. @lozanaross if you have any thoughts :)

Indeed, I think I should open a dedicated design issue - design a consistent mechanism to display a range of fail scenarios. I'll make it in the SDC repo and link here.

**Comment by connorshea** [comment]
Improvements to this would be really great, it's hard to tell if my reconciliation is even working while it's running sometimes, and that makes the tool a lot harder to use. It'd be good to know how many records its gone through so far, rather than just a % of progress.
